### PR TITLE
feat: improved duplication UI via modal

### LINKED ISSUE #560: "Duplicate prompt" UI improvements
#558 adds the ability to duplicate prompts. @thomasguillot has some good ideas on how to improve the UI to do so:

* When we click "Duplicate", should it open a modal so we can have the option to rename the prompt and display the category filtering to enable the user to change these
* Add a notice if the prompt will be "unassigned"
* Once the user clicks on the "Duplicate" button (Primary -- modal) they get a confirmation message with a link to edit it in Gutenberg?
* If they click "Cancel" (Secondary -- modal)  then nothing happens

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Improves the UI for duplicating prompts. When duplicating a prompt from the single-prompt edit screen, you are now presented with a modal with the opportunity to rename the duplicate before creating it, an option to close the modal or edit the duplicate after it's created, and a notice if the duplicate is created without a campaign (which would cause it to be visible in the Campaigns dashboard only under the "unassigned" filter).

Closes #560.

#### Campaigns dashboard UI

1. Go to **Newspack > Campaigns**. Open the primary popover menu and select "Duplicate". Confirm that you now see a modal with a text input to rename the duplicate, which loads the default name ("Prompt title copy", "Prompt title copy 2", etc.) at first.

Screen Shot 2021-06-17 at 4 18 30 PM

2. Edit the name, but click "Cancel", click "Duplicate" again on the same prompt, confirm that the changes you had entered are discarded.
3. Click "Cancel" again and click "Duplicate" for another prompt. Confirm that the modal now shows info for whatever prompt you clicked "Duplicate" for.
4. Duplicate the prompt. Confirm that after successful duplication, you are presented with a success message and the option to close the modal (remaining on the Campaigns dashboard page) or edit the duplicate (which redirects you to the edit page for the newly created prompt). You can also test the failure state by blocking the request in the Dev Tools > Network tab. Note that when blocking a request you must attempt to duplicate the same prompt and with the same title as when blocked. A failed duplication request should result in the standard "fatal error" modal for all Newspack dashboard pages.

Screen Shot 2021-06-17 at 4 24 07 PM

5. Edit the duplicate and confirm that the title of the duplicate matches whatever you entered in the input field before, and that the content, taxonomy terms, and prompt settings all match those of the prompts you duplicated.
6. Create another duplicate, but make sure you duplicate a prompt that's not assigned. Confirm that both before and after you create the duplicate, the modal shows a warning that the prompt is unassigned.

Screen Shot 2021-06-18 at 2 37 55 PM

7. Also test that if you create multiple duplicates of the same prompt (or duplicates of the duplicates of a prompt), the default title increments as expected.

Screen Shot 2021-06-17 at 4 24 18 PM

#### Single prompt edit UI

1. Edit any prompt. Click the "Duplicate" button next to the "Preview" button.
2. The modal UI should be pretty much exactly the same as from the Campaigns dashboard. However, we're using core WP components here instead of Newspack components because when I tried using Newspack components, I kept running into [this same issue](https://github.com/Automattic/newspack-popups/issues/556) that happens when trying to preview. @thomasguillot or @adekbadek, do you have any ideas about why the `@wordpress/icons` package seems to not be importable when Newspack Components are imported as an NPM package? If we can solve this bug, we can switch the components here to Newspack Components instead of core.

Screen Shot 2021-06-18 at 2 42 21 PM

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by thomasguillot — APPROVED** [review]
@dkoo, I pushed a few changes: https://github.com/Automattic/newspack-popups/pull/564/commits/e8a5cf81b446b685768f4325b52bec0bc35a7e88

**Comment by thomasguillot** [comment]
> do you have any ideas about why the @wordpress/icons package seems to not be importable when Newspack Components are imported as an NPM package? If we can solve this bug, we can switch the components here to Newspack Components instead of core.

I don't know why it crashes but we should be using Core components in the editor anyway.

**Comment by dkoo** [comment]
@thomasguillot Changes look good, thanks!

File: src/editor/Duplicate.js [hunk 1]
```diff
@@ -2,58 +2,168 @@
...
+								<Button isPrimary href={ `/wp-admin/post.php?post=${ duplicated }&action=edit` }>
+									{ __( 'Edit', 'newspack-popups' ) }
+								</Button>
+							</Flex>
+						</>
+					) : (
+						<>
+							{ ( ! campaignGroups || 0 === campaignGroups.length ) && (
+								<Notice status="warning" isDismissible={ false }>
+									{ __( 'This prompt will be unassigned.', 'newspack-popups' ) }
```
Comment (adekbadek) [inline src/editor/Duplicate.js:117]: Just to make it more understandable:

```suggestion
									{ __( 'The new prompt will no be assigned to any campaign.', 'newspack-popups' ) }
```

File: src/editor/Duplicate.js [hunk 2]
```diff
@@ -2,58 +2,168 @@
...
+						<>
+							<Notice status="success" isDismissible={ false }>
+								{ sprintf(
+									__( 'Duplicate of “%s” created as a draft.', 'newspack-popups' ),
+									title
+								) }
+							</Notice>
+							{ ( ! campaignGroups || 0 === campaignGroups.length ) && (
+								<Notice status="warning" isDismissible={ false }>
+									{ __( 'The prompt is currently unassigned.', 'newspack-popups' ) }
```
Comment (adekbadek) [inline src/editor/Duplicate.js:101]: WDYT think about demystifying "unassigned" a bit:

```suggestion
									{ __( 'This prompt is currently not assigned to any campaign.', 'newspack-popups' ) }
```

**Review by adekbadek — CHANGES_REQUESTED** [review]

File: src/editor/Duplicate.js (hunk 1 above)
Comment (dkoo) [reply src/editor/Duplicate.js]: Good suggestion! Updated in d40359b.

**Review by adekbadek — APPROVED** [review]
